src/Notifier.cpp

```diff
@@ -163,9 +183,9 @@ size_t Notifier::ReceiveConfirmation() {
  * Reset the Shotgun-Alien queue to nullptr
  */
 void Notifier::Reset() {
-   std::lock_guard<std::mutex> guard(gLock);
    gQueue.reset(nullptr);
    gHandshakeQueue.reset(nullptr);
+   mKeepRunning.store(false);
```

Comment:
the mKeepRunning is so that the notifier  can stop the loop relatively quickly after destruction. without adding the mKeepRunning to the conditional, I HAVE to wait timeout seconds before  the notifier closes.

i tested this...it works